File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-5.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int left = 0;
        int right = nums.size()-1;
        if (nums[0] <= nums[nums.size()-1]) return nums[0]; //sorted
        
        //when left == right the answer is found
        while (left < right) {
            int mid = (left+right)/2;
            
            if (nums[mid] >= nums[right]) //min is at the right side, not including mid
                left = mid+1;
            else //min is at the left side may include mid
                right = mid;

        }

        return nums[left];
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-5.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        #dictionary

        count = dict()

        for n in nums:
            count[n] = count.get(n,0) + 1
            if (count[n] > 1):
                return True
        return False

        
        
        